packages/json/src/lib/JSONProvider.ts

```diff
@@ -0,0 +1,693 @@
+      }
+    }
+
+    return payload;
+  }
+
+  public async [Method.Get]<Value = StoredValue>(payload: GetPayload<Value>): Promise<GetPayload<Value>> {
+    const { key, path } = payload;
+    const value = await this.handler.get(key);
+
+    Reflect.set(payload, 'data', path.length === 0 ? value : getFromObject(value, path));
+
+    return payload;
+  }
+
+  public async [Method.GetAll](payload: GetAllPayload<StoredValue>): Promise<GetAllPayload<StoredValue>> {
+    for (const [key, value] of await this.handler.entries()) payload.data[key] = value;
+
+    return payload;
+  }
+
+  public async [Method.GetMany](payload: GetManyPayload<StoredValue>): Promise<GetManyPayload<StoredValue>> {
+    const { keys } = payload;
+
+    for (const key of keys) payload.data[key] = (await this.handler.get(key)) ?? null;
+
+    return payload;
+  }
+
+  public async [Method.Has](payload: HasPayload): Promise<HasPayload> {
+    const { key, path } = payload;
+
+    if (await this.handler.has(key)) {
+      payload.data = true;
+
+      if (path.length !== 0) payload.data = hasFromObject(await this.handler.get(key), path);
+    }
+
+    return payload;
+  }
+
+  public async [Method.Inc](payload: IncPayload): Promise<IncPayload> {
+    const { key, path } = payload;
+    const { data } = await this.get({ method: Method.Get, key, path });
+
+    if (data === undefined) {
+      payload.error = new JSONProviderError({
+        identifier: JSONProvider.CommonIdentifiers.IncMissingData,
+        message: path.length === 0 ? `The data at "${key}" does not exist.` : `The data "${key}.${path.join('.')}" does not exist.`,
+        method: Method.Inc
+      });
+
+      return payload;
+    }
+
+    if (typeof data !== 'number') {
+      payload.error = new JSONProviderError({
+        identifier: JSONProvider.CommonIdentifiers.IncInvalidType,
+        message:
+          path.length === 0 ? `The data at "${key}" must be of type "number".` : `The data at "${key}.${path.join('.')}" must be of type "number".`,
+        method: Method.Inc
+      });
+
+      return payload;
+    }
+
+    await this.set({ method: Method.Set, key, path, value: data + 1 });
+
+    return payload;
+  }
+
+  public async [Method.Keys](payload: KeysPayload): Promise<KeysPayload> {
+    payload.data = await this.handler.keys();
+
+    return payload;
+  }
+
+  public async [Method.Map]<DataValue = StoredValue, HookValue = DataValue>(
+    payload: MapByHookPayload<DataValue, HookValue>
+  ): Promise<MapByHookPayload<DataValue, HookValue>>;
+
+  public async [Method.Map]<DataValue = StoredValue>(payload: MapByPathPayload<DataValue>): Promise<MapByPathPayload<DataValue>>;
+  public async [Method.Map]<DataValue = StoredValue, HookValue = DataValue>(
+    payload: MapPayload<DataValue, HookValue>
+  ): Promise<MapPayload<DataValue, HookValue>> {
+    if (isMapByHookPayload(payload)) {
+      const { hook } = payload;
+
+      // @ts-expect-error 2345
```

Comment:
Could a reason be provided for these

Reply:
The reason is given in a TypeScript error code. Which is how we've been doing all `@ts-expect-error`s

Reply:
Yea but that code hasn't been mentioned in 5 years after a quick google haha

Reply:
A list of them can be found [here](https://github.com/Microsoft/TypeScript/blob/v4.5.5/src/compiler/diagnosticMessages.json)